src/ImageSharp/Formats/Dds/DdsDecoderCore.cs

```diff
@@ -0,0 +1,190 @@
+        public Texture DecodeTexture(Stream stream)
+        {
+            try
+            {
+                this.ReadFileHeader(stream);
+
+                if (this.ddsHeader.Width == 0 || this.ddsHeader.Height == 0)
+                {
+                    throw new UnknownImageFormatException("Width or height cannot be 0");
+                }
+
+                if (this.ddsHeader.IsVolumeTexture())
+                {
+                    var depths = ddsHeader.ComputeDepth();
+
+                    var texture = new Texture(TextureType.VolumeTexture)
+                    {
+                        Images = new Image[depths][]
+                    };
+
+                    for (int depth = 0; depth < depths; depth++)
+                    {
+                        texture.Images[depth] = Processing.Dds.DecodeDds(stream, this.ddsHeader, this.ddsDxt10header);
+                    }
+
+                    return texture;
+                }
+                else if (this.ddsHeader.IsCubemap())
+                {
+                    DdsSurfaceType[] faces = this.ddsHeader.GetExistingCubemapFaces();
+                    var texture = new Texture(TextureType.VolumeTexture)
+                    {
+                        Images = new Image[faces.Length][]
+                    };
+
+                    for (int face = 0; face < faces.Length; face++)
+                    {
+                        texture.Images[face] = Processing.Dds.DecodeDds(stream, this.ddsHeader, this.ddsDxt10header);
+                    }
+
+                    return texture;
+                }
+                else
+                {
+                    var texture = new Texture(TextureType.VolumeTexture)
+                    {
+                        Images = new Image[1][]
+                    };
+
+                    texture.Images[0] = Processing.Dds.DecodeDds(stream, this.ddsHeader, this.ddsDxt10header);
+
+                    return texture;
+                }
+            }
+            catch (IndexOutOfRangeException e)
+            {
+                throw new ImageFormatException("Dds image does not have a valid format.", e);
+            }
+        }
+
+        /// <summary>
+        /// Reads the raw image information from the specified stream.
+        /// </summary>
+        /// <param name="stream">The <see cref="Stream"/> containing image data.</param>
+        public IImageInfo Identify(Stream stream)
+        {
+            this.ReadFileHeader(stream);
+
+            D3dFormat d3dFormat = this.ddsHeader.PixelFormat.GetD3DFormat();
+            DxgiFormat dxgiFormat = this.ddsHeader.ShouldHaveDxt10Header() ? this.ddsDxt10header.DxgiFormat : DxgiFormat.Unknown;
+            int bitsPerPixel = DdsTools.GetBitsPerPixel(d3dFormat, dxgiFormat);
+
+            return new ImageInfo(
+                new PixelTypeInfo(bitsPerPixel),
+                (int)this.ddsHeader.Width,
+                (int)this.ddsHeader.Height,
+                null);
```

Comment:
You wanna make sure this provides as much format specific information as you can.